### PR TITLE
Feat: Allows dynamic sources

### PR DESCRIPTION
**Pull request by viwell**

### CODE REVIEW

File: src/auto-complete.component.ts [hunk 1]
```diff
@@ -124,6 +124,7 @@ export class NguiAutoCompleteComponent implements OnInit {
   @Input("show-dropdown-on-init") showDropdownOnInit: boolean = false;
   @Input("tab-to-select") tabToSelect: boolean = true;
   @Input("match-formatted") matchFormatted: boolean = false;
+  @Input("source-changes") sourceChanges: EventEmitter<any>;
```
Comment (allenhwkim) [inline src/auto-complete.component.ts]: Isn't it @Output?

File: src/auto-complete.directive.ts [hunk 2]
```diff
@@ -129,8 +131,11 @@ export class NguiAutoCompleteDirective implements OnInit {
   }
 
   ngOnChanges(changes: SimpleChanges): void {
-    if (changes['ngModel']) {
-      this.ngModel = this.setToStringFunction(changes['ngModel'].currentValue);
+    // if (changes['ngModel']) {
+    //   this.ngModel = this.setToStringFunction(changes['ngModel'].currentValue);
+    // }
+    if (changes["source"]) {
+      this.sourceChanges.emit(this.source);
```
Comment (allenhwkim) [inline src/auto-complete.directive.ts]: I am not sure why this is commented out?

File: src/auto-complete.component.ts (hunk 1 above)
Comment (allenhwkim) [inline src/auto-complete.component.ts]: The name has better to be `source-changed`, I think

Comment (viwell) [inline src/auto-complete.component.ts]: For any reason the source variable is not updating its value, so I've forced to do so using an event emitter who listen to external changes, that's why is an input.

File: src/auto-complete.directive.ts (hunk 2 above)
Comment (viwell) [inline src/auto-complete.directive.ts]: This class wasn't implementing the OnChanges lifecycle then those lines weren't fired, so in order to not break anything they were commented out.

**Comment by allenhwkim** [comment]
This PR might be necessary if [src] change is acknowledged.

https://stackoverflow.com/a/34694155/454252

File: src/auto-complete.directive.ts (hunk 2 above)
Comment (GeorgeSokolovsky) [inline src/auto-complete.directive.ts]: Actually, if class doesn't implement lifecycle hook interface, it doesn't mean that this method doesn't work. Do not forget that interfaces exists only at TS code, so at runtime Angular will call this method as expected.

File: src/auto-complete.component.ts (hunk 1 above)
Comment (GeorgeSokolovsky) [inline src/auto-complete.component.ts]: I prefer to use ```EventEmitter``` only for ```@Output``` as it was designed.

File: src/auto-complete.component.ts [hunk 3]
```diff
@@ -314,4 +320,11 @@ export class NguiAutoCompleteComponent implements OnInit {
     };
   })();
 
+  private subscribeToSourceChanges(): void {
+    this.sourceChanges.subscribe((value) => {
```
Comment (GeorgeSokolovsky) [inline src/auto-complete.component.ts]: I suppose to unsubscribe at ```ngOnDestroy```

**Pull request closed without merging**